src/modules/filelist/virtualized/cells/Cell.jsx

```diff
@@ -0,0 +1,183 @@
+import { useVaultClient } from 'cozy-keys-lib'
+import { useSharingContext } from 'cozy-sharing'
+import { useBreakpoints } from 'cozy-ui/transpiled/react/providers/Breakpoints'
+import { useI18n } from 'cozy-ui/transpiled/react/providers/I18n'
+
+import AcceptingSharingContext from '@/lib/AcceptingSharingContext'
+import { ActionMenuWithHeader } from '@/modules/actionmenu/ActionMenuWithHeader'
+import {
+  isRenaming as isRenamingSelector,
+  getRenamingFile
+} from '@/modules/drive/rename'
+import AddFolder from '@/modules/filelist/AddFolder'
+import FileOpener from '@/modules/filelist/FileOpener'
+import FileAction from '@/modules/filelist/virtualized/cells/FileAction'
+import FileName from '@/modules/filelist/virtualized/cells/FileName'
+import LastUpdate from '@/modules/filelist/virtualized/cells/LastUpdate'
+import Share from '@/modules/filelist/virtualized/cells/Share'
+import Size from '@/modules/filelist/virtualized/cells/Size'
+import { useSelectionContext } from '@/modules/selection/SelectionProvider'
+import { isReferencedByShareInSharingContext } from '@/modules/views/Folder/syncHelpers'
+
+const Cell = ({
+  row,
+  column,
+  cell,
+  currentFolderId,
+  withFilePath,
+  actions
+}) => {
+  const { f, t } = useI18n()
+  const vaultClient = useVaultClient()
+
+  const { isExtraLarge } = useBreakpoints()
+  const { sharingsValue } = useContext(AcceptingSharingContext)
+  const { byDocId } = useSharingContext()
+  const filerowMenuToggleRef = useRef()
+  const { toggleSelectedItem, isSelectionBarVisible } = useSelectionContext()
+  const [showActionMenu, toggleShowActionMenu] = useReducer(
+    state => !state,
+    false
+  )
+  const isRenaming = useSelector(
+    state =>
+      isRenamingSelector(state) && get(getRenamingFile(state), 'id') === row.id
+  )
+
+  if (row.type === 'tempDirectory') {
+    if (column.id === 'name') {
+      return (
+        <AddFolder
+          vaultClient={vaultClient}
+          currentFolderId={currentFolderId}
+        />
+      )
+    }
+
+    if (column.id === 'menu') {
+      return null
+    }
+
+    return '—'
+  }
+
+  const formattedSize =
+    !isDirectory(row) && row.size ? filesize(row.size, { base: 10 }) : undefined
+  const updatedAt = row.updated_at || row.created_at
+  const formattedUpdatedAt = f(
+    updatedAt,
+    isExtraLarge
+      ? t('table.row_update_format_full')
+      : t('table.row_update_format')
+  )
+  const isSharingContextEmpty = Object.keys(sharingsValue).length <= 0
+  const isInSyncFromSharing =
+    !isSharingContextEmpty &&
+    isSharingShortcut(row) &&
+    isReferencedByShareInSharingContext(row, sharingsValue)
+  const isRowDisabledOrInSyncFromSharing = isInSyncFromSharing
+  // We don't allow any action on shared drives and trash
+  // because they are magic folder created by the stack
+  const canInteractWithFile =
+    row._id !== 'io.cozy.files.shared-drives-dir' &&
+    !row._id.endsWith('.trash-dir')
```

Comment:
nit: Maybe we can move some of this condition inside the relevant if (like isInSyncFromSharing could be below `column.id === 'share'` condition

Reply:
moved canInteractWithFile inside `column.id === 'menu'` 👍 

`isInSyncFromSharing` is used on two places